source/common/network/io_socket_handle_impl.h

```diff
@@ -12,13 +12,29 @@
 namespace Envoy {
 namespace Network {
 
+/**
+ * On different platforms the sockaddr struct for unix domain
+ * sockets is different. We use this function to get the
+ * length of the platform specific struct.
+ */
+static constexpr socklen_t udsAddressLength() {
+#if defined(__APPLE__)
+  return sizeof(sockaddr);
+#elif defined(WIN32)
+  return sizeof(sockaddr_un);
+#else
+  return sizeof(sa_family_t);
+#endif
+}
+
 /**
  * IoHandle derivative for sockets.
  */
 class IoSocketHandleImpl : public IoHandle, protected Logger::Loggable<Logger::Id::io> {
 public:
-  explicit IoSocketHandleImpl(os_fd_t fd = INVALID_SOCKET, bool socket_v6only = false)
-      : fd_(fd), socket_v6only_(socket_v6only) {}
+  explicit IoSocketHandleImpl(os_fd_t fd = INVALID_SOCKET, bool socket_v6only = false,
```

Comment:
This should probably not be defaulted and it should be a required parameter.

Reply:
Since many tests inject fake fds, I kept the same signature and added `absl::nullopt` as the default